### DISCUSSION TITLE
Using zig as a clang compiler

### DISCUSSION THREAD

**Discussion by SebastianMpl** [vlang/v#13773] [General]
`Hey` guys,
I'm trying to use zig which is clang compatibile c compiler with multiple libc and ability to compile for other operating systems.

I made an appimage that points to zig cc and did a symlink named clang to /usr/bin but I get error:

```
[seba@sebapc v]$ ./v -cc clang ./structtest.v 
==================
Unable to hash /home/seba/Obrazy/v/"/tmp/v_1000/structtest.13441100194319091039.tmp.c": file not found
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

I even renamed clang to zig and it's still the same.
I know that zig is not officially supported, but is there any way to make it work?
Anyone tried zig cc with vlang before?

Thank you

**Comment by ylluminate** [comment]
While not commenting directly to the actual question: this is probably an Issue vs Discussion item since this, on the outset, does appear to be some kind of a bug that we need to look into further...

**Comment by free1139** [comment]
I make a shell named 'zc' it works.
```
#!/bin/sh
zig cc $@
```
then
```
v -cc "zc" -showcc -show-c-output .
```

but do a cross compile is researching.

**Comment by JalonSolov** [comment]
I don't know the current state of it, but https://github.com/malisipi/vzcc was created a while ago.

**Reply by Wajinn** [reply]
malisipi's [vzcc](https://github.com/malisipi/vzcc) is for the exact purpose. Looks like those commenting should try it out and if any issues, contact him about it.
